02_knight_and_princess/roman_kuprii/main.cpp

```diff
@@ -17,13 +19,35 @@ using VertexId = int;
 constexpr double GREEN_TRASHOULD = 0.1;
 constexpr double BLUE_TRASHOULD = 0.25;
 constexpr double RED_TRASHOULD = 0.33;
+constexpr int GRAPH_NUMBER = 10;
+constexpr int COLORS_NUMBER = 5;
 constexpr int INVALID_ID = -1;
 constexpr int INVALID_NEW_DEPTH = -1;
 constexpr int INVALID_NEW_VERTICES_NUM = -1;
-const std::string JSON_GRAPH_FILENAME = "graph.json";
+const std::string JSON_GRAPH_FILENAME = "temp/graph_";
+const std::string LOG_FILENAME = "log.txt";
 
 enum class Color { Gray, Green, Blue, Yellow, Red };
 
+class Logger {
+ public:
+  static Logger& get_logger() {
+    static Logger logger;
+    return logger;
+  }
+
+  void write(std::ofstream& logfile, const std::string log) const {
```

Comment:
1. `log` - это действие. Называть так строку не правильно, а вот что правильно - назвать так метод.
2. Каждый раз передавать файл - это оверкилл. Укажи его один раз и пусть он помнит куда 
логировать.
3. Почему `const` значение, а не ссылка?
```suggestion
  void log(const std::string& string) const {
```

Reply:
По поводу 2 пункта - std::ofstream& logfile это объект, чтобы не передавать в логгер его нужно в нём сохранить тогда, это укладывается в логику логгера?